.github/workflows/release-pypi.yml

```diff
@@ -7,48 +7,171 @@ on:  # Trigger the workflow on push or pull request, but only for the master bra
   release:
     types: [published]
 
+# there are several consecutive actions:
+# 1) determine which packages have been change at the time this event is processed
+# 2) build related packages - app/pytorch or download latest from pypi
+# 3) create the meta package - lightning
+# 4) publish all new creations tada
 
 jobs:
-  # Todo run job which determine changed versions
+  # run job which determine changed versions
+  releasing:
+    runs-on: ubuntu-20.04
+    outputs:
+      build-pkgs: ${{ steps.candidate.outputs.pkgs }}
+      pull-pkgs: ${{ steps.download.outputs.pkgs }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - run: |
+        pip install -q fire requests
+        mkdir dist && touch dist/.placeholder
+        mkdir pypi && touch pypi/.placeholder
+    - uses: actions/upload-artifact@v3
+      with:
+        name: dist-packages-${{ github.sha }}
+        path: dist
+    - uses: actions/upload-artifact@v3
+      with:
+        name: pypi-packages-${{ github.sha }}
+        path: pypi
+
+    - name: Find changed packages
+      id: candidate
+      run: |
+        echo "::set-output name=pkgs::{include: $(python .actions/assistant.py determine-releasing-pkgs 2>&1)}"
+    - run: echo "${{ steps.candidate.outputs.pkgs }}"
+
+    - name: Inverse packages to pull
+      id: download
+      run: |
+        echo "::set-output name=pkgs::{include: $(python .actions/assistant.py determine-releasing-pkgs --inverse 2>&1)}"
+    - run: echo "${{ steps.download.outputs.pkgs }}"
 
   # based on https://github.com/pypa/gh-action-pypi-publish
   build-package:
-    # todo run sequential
+    needs: releasing
     runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: true
+      # run sequential
+      max-parallel: 1
+      matrix: ${{ fromJSON(needs.releasing.outputs.build-pkgs) }}
     steps:
     - uses: actions/checkout@v2
+    - uses: actions/download-artifact@v3
+      with:
+        name: dist-packages-${{ github.sha }}
+        path: dist
     - uses: actions/setup-python@v2
       with:
         python-version: 3.9
 
     - name: Install dependencies
-      run: pip install --upgrade setuptools wheel
+      run: pip install -U setuptools wheel
 
     - name: Build packages
-      # TODO: this would be big challenge as at the release time (especially from master)
-      #  all packages from meta is generated wont be in stable mode,
-      #  so the meta will be for one as release and other as dev
-      # todo: for meta pkg generation we need to install the unstable pkg from PyPI and overwrite source
       env:
-        PACKAGE_NAME: pytorch
+        PACKAGE_NAME: ${{ matrix.pkg }}
       run: |
         python setup.py sdist bdist_wheel
         ls -lh dist/
 
-    - uses: actions/upload-artifact@v2
+    - uses: actions/upload-artifact@v3
+      with:
+        name: dist-packages-${{ github.sha }}
```

Comment:
Should this contain the package name also? 
In case both `app` and `pytorch` are updated in the same PR, it will be overridden.

Reply:
in such case we may have hard time to aggregate all for later release